### PR TITLE
`dart pub outdated` - Better message when no resolution exists.

### PR DESCRIPTION
**Pull request by sigurdm**
Fixes: https://github.com/dart-lang/pub/issues/2833

### CODE REVIEW

File: lib/src/command/outdated.dart [hunk 1]
```diff
@@ -421,17 +430,16 @@ Future<void> _outputJson(
...
+Future<void> _outputHuman(List<_PackageDetails> rows, Mode mode,
+    {@required bool showAll,
+    @required bool useColors,
+    @required bool includeDevDependencies,
+    @required bool lockFileExists,
+    @required bool hasDirectDependencies,
+    @required bool hasDevDependencies,
+    @required bool showTransitiveDependencies,
+    @required bool hasUpgradableResolution,
+    @required bool hasResolvableResolution}) async {
```
Comment (jonasfj) [inline lib/src/command/outdated.dart:442]: Can I have trailing comma to keep alignment :D

File: lib/src/command/outdated.dart [hunk 2]
```diff
@@ -608,6 +615,10 @@ Showing outdated packages.
   @override
   String get allGood => 'all up-to-date.';
 
+  @override
+  String get noResolutionText =>
+      '''No resolution was found. Try running `dart pub upgrade --dry-run to explore why.''';
```
Comment (jonasfj) [inline lib/src/command/outdated.dart:620]:
```suggestion
      '''No resolution was found. Try running `dart pub upgrade --dry-run` to explore why.''';
```

File: lib/src/command/outdated.dart [hunk 3]
```diff
@@ -680,6 +691,10 @@ Showing dependencies that are currently not opted in to null-safety.
   @override
   String get allGood => 'all support null safety.';
 
+  @override
+  String get noResolutionText =>
+      '''No resolution was found. Try running `dart pub upgrade --null-safety --dry-run to explore why.''';
```
Comment (jonasfj) [inline lib/src/command/outdated.dart:696]:
```suggestion
      '''No resolution was found. Try running `dart pub upgrade --null-safety --dry-run` to explore why.''';
```

File: test/outdated/goldens/null_safety_no_resolution.txt [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+Showing dependencies that are currently not opted in to null-safety.
+[✗] indicates versions without null safety support.
+[✓] indicates versions opting in to null safety.
+
+Package Name  Current  Upgradable  Resolvable  Latest               
+
+direct dependencies:
+bar           ✗1.0.0   ✗1.0.0      -           ✓2.0.0-nullsafety.0  
+foo           ✗1.0.0   ✗1.0.0      -           ✓2.0.0-nullsafety.0  
+No resolution was found. Try running `dart pub upgrade --null-safety --dry-run to explore why.
```
Comment (jonasfj) [inline test/outdated/goldens/null_safety_no_resolution.txt:64]:
```suggestion
No resolution was found. Try running `dart pub upgrade --null-safety --dry-run` to explore why.
```

File: test/outdated/goldens/null_safety_no_resolution.txt [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
```
Comment (jonasfj) [inline test/outdated/goldens/null_safety_no_resolution.txt:76]:
```suggestion
No resolution was found. Try running `dart pub upgrade --null-safety --dry-run` to explore why.
```

File: test/outdated/goldens/null_safety_no_resolution.txt [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+Showing dependencies that are currently not opted in to null-safety.
+[✗] indicates versions without null safety support.
+[✓] indicates versions opting in to null safety.
+
+Package Name  Current  Upgradable  Resolvable  Latest  
+
+direct dependencies:
+bar           ✗1.0.0   ✗1.0.0      -           ✗1.0.0  
+foo           ✗1.0.0   ✗1.0.0      -           ✗1.0.0  
+No resolution was found. Try running `dart pub upgrade --null-safety --dry-run to explore why.
```
Comment (jonasfj) [inline test/outdated/goldens/null_safety_no_resolution.txt:88]:
```suggestion
No resolution was found. Try running `dart pub upgrade --null-safety --dry-run` to explore why.
```

**Review by jonasfj — APPROVED** [review]

File: lib/src/command/outdated.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/command/outdated.dart]: :) yes

File: lib/src/command/outdated.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/command/outdated.dart]: Done

File: lib/src/command/outdated.dart (hunk 3 above)
Comment (sigurdm) [reply lib/src/command/outdated.dart]: Done

**Pull request merged**